pyuvm/s18_pyuvm_reg_block.py

```diff
@@ -1,19 +1,182 @@
 from pyuvm import uvm_object
-#import vsc
+from pyuvm.s20_pyuvm_reg import uvm_reg
+from pyuvm.s21_pyuvm_reg_map import uvm_reg_map
+from pyuvm.s24_pyuvm_reg_includes import uvm_fatal, uvm_not_implemeneted
 
-# 18.1.1 Class declaration
-class pyuvm_reg_block(uvm_object):
 
-    # 18.1.2.1
-    # TODO Fix signature
-    def __init__(self, name=""):
+class uvm_reg_block(uvm_object):
+    def __init__(self, name="uvm_reg_block"):
         super().__init__(name)
         self._regs = []
+        self.def_map = None
+        self._is_locked = False
+        self.hdl_paths = {}
+        self.fields = []
+        self.root_path = ""
+        self.child_blk = []
+        self.maps = []
+        self.parent_blk = None
+        # optimize this by using dicationries with
+        # TUPLES (existence_bool, handle)
+        self.blk_maping = {}
+        self.map_mapping = {}
+        self.reg_mapping = {}
+        self.blk_name = name
+        self.header = name + " -- "
```

Comment:
flake8 fixes, TODOs added (will soon remove the Mapping Dictionary and replaced with TUPLES)